src/CachePlugin.php

```diff
@@ -160,6 +163,22 @@ private function getMaxAge(ResponseInterface $response)
             return (new \DateTime($header))->getTimestamp() - (new \DateTime())->getTimestamp();
         }
 
-        return $this->defaultTtl;
+        return $this->config['default_ttl'];
+    }
+
+    /**
+     * @param array $config
+     *
+     * @return array
+     */
+    protected function configure(array $config = [])
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults($this->config);
```

Comment:
When the config array is in the constructor, there is no issue. But I'll change this when I move it a setter. 
Thank you for the feedback.